### ISSUE TITLE
Random Registration error 

### ISSUE THREAD

**Issue by sairajkishore** [ibm-bluemix-mobile-services/bms-clientsdk-android-push#80]
Hi,
Am getting below mentioned error while registering with blumix, its not every time reproducible but randomly happening. could you let me know what woulb be the root cause?

com.ibm.mobilefirstplatform.clientsdk.android.push.api.a: , Message : {"code":"FPWSE0005E","message":"Invalid value was provided. Check the 'token' parameter value."}, statusCode : 400

**Comment by AnanthaKrish** [comment]
@sairajkishore Hi, Seems like you are not getting token from the FCM. Do you have a valid app in FCM and added the app and push SDK bundle identifier there ?.... Please check the `google-services.json` file.....Make sure you are following the documents.......

Let me know if you need help.... thanks

**Comment by sairajkishore** [comment]
Yes i have valid google-service.json as it works for most of the time.only in few occasions below mention error comes.

com.ibm.mobilefirstplatform.clientsdk.android.push.api.a: , Message : {"code":"FPWSE0005E","message":"Invalid value was provided. Check the 'token' parameter value."}, statusCode : 400

**Comment by AnanthaKrish** [comment]
@sairajkishore That looks strange ... I will try to recreate it and let you know ..

**Comment by AnanthaKrish** [comment]
@sairajkishore this is happening when you register your app for push notifications ?

**Comment by sairajkishore** [comment]
yes it is  happening when registration happening for push notification.

**Comment by AnanthaKrish** [comment]
@sairajkishore are you using the emulator to run the app ? This is happening due to the delay in connection with Firebase. If there is a delay the master token , which is null most of the time, will be assigned to the token value. By the way I will handle this token null condition.

**Comment by sairajkishore** [comment]
Am not using any emulator ,it is a production build only.

**Comment by AnanthaKrish** [comment]
@sairajkishore are you able to get the value for this log `MFPPush:registerInBackground() - Successfully registered with FCM. Returned deviceToken is:` ? If yes can you please share that ? If you have the logs for this it will help us figure out the exact issue ...

**Comment by sairajkishore** [comment]
Hi,
I could not have any logs, it not happening every time, but randomly happening the logs which i got is from production.
